### DISCUSSION TITLE
Router Configuration in react

### DISCUSSION THREAD

**Discussion by husainasfak** [remix-run/remix#7978] [General]
I am using the new version of the remix. how to use the folder-based route in the new version?

**Comment by xiamuguizhi** [comment]
这些文档能帮助到您。

https://remix.run/docs/en/main/file-conventions/routes

https://interactive-remix-routing-v2.netlify.app/

https://remix.run/docs/en/main/file-conventions/routes#nested-routes

**Reply by husainasfak** [reply]
Thanks for the quick reply. I have one more question which router configuration is best for scaling or a build large app. dot(.) notations route config or folder-based config

**Reply by xiamuguizhi** [reply]
Dot notation route config: This approach involves defining routes using a dot notation format, where each route is represented as a string path. This approach can be suitable for smaller projects or when you have a relatively flat route structure. It offers simplicity and ease of maintenance, as all routes are defined in a single file. However, as the app grows larger, managing a complex route structure using a single file can become challenging.

Folder-based config: In this approach, routes are organized into separate files based on the folder structure of your app. Each folder represents a route or a group of related routes, and the configuration is spread across multiple files. This approach provides better scalability and modularity, as you can easily add, remove, or modify routes without affecting the entire configuration. It also allows for better organization and separation of concerns, making it easier to navigate and maintain the codebase as it grows.
